src/mail/view/ConversationViewModel.ts

```diff
@@ -0,0 +1,257 @@
+				const newSubject = this.normalizeSubject(mail.subject)
+				let index = findLastIndex(conversation, (i) => firstBiggerThanSecond(getElementId(entry), elementIdPart(i.entryId)))
+				if (index < 0) {
+					index = conversation.length
+				}
+				const lastSubject = findLast(conversation, (c) => c.type === "subject") as SubjectItem | null
+				if (newSubject !== lastSubject?.subject) {
+					conversation.splice(index + 1, 0, { type: "subject", subject: newSubject, id: getElementId(mail), entryId: entry._id })
+					index += 1
+				}
+				conversation.splice(index + 1, 0, { type: "mail", viewModel: this.viewModelFactory({ ...this.options, mail }), entryId: entry._id })
+				this.onUiUpdate()
+			}
+		} catch (e) {
+			if (e instanceof NotFoundError) {
+				// Ignore, something was already deleted
+			} else {
+				throw e
+			}
+		}
+	}
+
+	private async processUpdateConversationEntry(update: EntityUpdateData) {
+		try {
+			// first wait that we load the conversation, otherwise we might already have the email
+			await this.loadingPromise
+		} catch (e) {
+			return
+		}
+		const conversation = assertNotNull(this.conversation)
+		const ceId: IdTuple = [update.instanceListId, update.instanceId]
+		let conversationEntry: ConversationEntry
+		let mail: Mail | null
+		try {
+			conversationEntry = await this.entityClient.load(ConversationEntryTypeRef, ceId)
+			mail =
+				// ideally checking the `mail` ref should be enough but we sometimes get an update with UNKNOWN and non-existing email but still with the ref
+				conversationEntry.conversationType !== ConversationType.UNKNOWN && conversationEntry.mail
+					? await this.entityClient.load(MailTypeRef, conversationEntry.mail)
+					: null
+		} catch (e) {
+			if (e instanceof NotFoundError) {
+				// Ignore, something was already deleted
+				return
+			} else {
+				throw e
+			}
+		}
+
+		const oldItemIndex = conversation.findIndex(
+			(e) => (e.type === "mail" && isSameId(e.viewModel.mail.conversationEntry, ceId)) || (e.type === "deleted" && isSameId(e.entryId, ceId)),
+		)
+		if (oldItemIndex === -1) {
+			return
+		}
+		const oldItem = conversation[oldItemIndex]
+		if (mail && oldItem.type === "mail" && haveSameId(oldItem.viewModel.mail, mail)) {
+			console.log("Noop entry update?", oldItem.viewModel.mail)
+			// nothing to do really, why do we get this update again?
+		} else {
+			if (oldItem.type === "mail") {
+				oldItem.viewModel.dispose()
+			}
+			if (mail) {
+				conversation[oldItemIndex] = {
+					type: "mail",
+					viewModel: this.viewModelFactory({ ...this.options, mail }),
+					entryId: conversationEntry._id,
+				}
+			} else {
+				// When DELETED conversation status type is added, replace entry with deleted entry instead of splicing out
+				conversation.splice(oldItemIndex, 1)
+			}
+		}
+	}
+
+	private conversationListId() {
+		return listIdPart(this._primaryViewModel.mail.conversationEntry)
+	}
+
+	private async loadConversation() {
```

Comment:
are we loading the whole conversation before rendering anything? Can we get some placeholder ui for this and maybe use an async generator to show the selected mail & the ones closest to it asap while continuing to load in the background?

Reply:
We do show it, see `entries()`

Reply:
I think this ties into the default value for `.conversiation`. The placeholder UI and moving away from the spinny thing is also something for another time, maybe